app/services/charges/charge_models/custom_service.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Charges
+  module ChargeModels
+    class CustomService < Charges::ChargeModels::BaseService
+      protected
+
+      INITIAL_STATE = { total_units: BigDecimal('0'), amount: BigDecimal('0') }.freeze
+
+      def compute_amount
+        # TODO: fetch cached result to reinject it into the aggregation result as a "previous state"
+        # NOTE: cached result should take the timestamp into account to allow recompute from a specific datetime
+
+        # NOTE: load the custom aggregation logic
+        instance_eval(custom_aggregator)
+
+        aggregation = aggregator.per_event_aggregation.event_aggregation.each_with_object(INITIAL_STATE.dup) do |event, agg|
```

Comment:
Layout/LineLength: Line is too long. [124/120] (https://rubystyle.guide#max-line-length)